diff --git a/node_modules/expo-video/build/VideoPlayer.js b/node_modules/expo-video/build/VideoPlayer.js
index 0cb6687..029450f 100644
--- a/node_modules/expo-video/build/VideoPlayer.js
+++ b/node_modules/expo-video/build/VideoPlayer.js
@@ -8,6 +8,9 @@ import NativeVideoModule from './NativeVideoModule';
 export function useVideoPlayer(source, setup) {
     const parsedSource = typeof source === 'string' ? { uri: source } : source;
     return useReleasingSharedObject(() => {
+        if (parsedSource === null) {
+            return null;
+        }
         const player = new NativeVideoModule.VideoPlayer(parsedSource);
         setup?.(player);
         return player;
diff --git a/node_modules/expo-video/src/VideoPlayer.tsx b/node_modules/expo-video/src/VideoPlayer.tsx
index 9388d38..350bacd 100644
--- a/node_modules/expo-video/src/VideoPlayer.tsx
+++ b/node_modules/expo-video/src/VideoPlayer.tsx
@@ -15,6 +15,9 @@ export function useVideoPlayer(
   const parsedSource = typeof source === 'string' ? { uri: source } : source;
 
   return useReleasingSharedObject(() => {
+    if (parsedSource === null) {
+      return undefined;
+    }
     const player = new NativeVideoModule.VideoPlayer(parsedSource);
     setup?.(player);
     return player;
